### PR TITLE
470 fak video non mandatory

### PR DESCRIPTION
**Pull request by wbaccinelli**
Fixes https://github.com/PerfectFit-project/virtual-coach-issues/issues/470
Fixes https://github.com/PerfectFit-project/testing-tickets/issues/87

- The fsm goes directly to the execution state in case the quit date has been reached, even if the first aid kit video has not been completed
- Save the status when running it, to make sure that the correct state has been stored

### CODE REVIEW

File: scheduler/state_machine/state_machine_utils.py [hunk 1]
```diff
@@ -765,6 +765,28 @@ def retrieve_tracking_day(user_id: int, current_date: date) -> int:
...
+                .filter(UserStateMachine.users_nicedayuid == user_id)
+                .first())
+
+    # if a machine already exists, use the same primary key to update the row
+    if user_fsm is not None:
+        user_fsm.state = state
+
+        session.merge(user_fsm)
+        session.commit()
+
```
Comment (bscheltinga) [inline scheduler/state_machine/state_machine_utils.py:788]: Don't you need the `session.close()` here?

File: scheduler/state_machine/controller.py [hunk 2]
```diff
@@ -291,6 +294,9 @@ def on_dialog_rescheduled(self, dialog, new_date):
                           planned_date=new_date,
                           phase=1)
 
+    def on_new_day(self, current_date: date):
+        self.check_if_end_date(current_date)
+
```
Comment (bscheltinga) [inline scheduler/state_machine/controller.py:299]: When it this called?

**Review by bscheltinga — APPROVED** [review]

File: scheduler/state_machine/state_machine_utils.py (hunk 1 above)
Comment (wbaccinelli) [reply scheduler/state_machine/state_machine_utils.py:788]: I absolutely do, good catch!

File: scheduler/state_machine/controller.py (hunk 2 above)
Comment (wbaccinelli) [reply scheduler/state_machine/controller.py:299]: This is triggered when a `new_day` event is sent by the FSM. This happens every morning.

Comment (bscheltinga) [reply scheduler/state_machine/controller.py:299]: Check, I see it now in state_machine.py :)

File: scheduler/state_machine/state_machine_utils.py (hunk 1 above)
Comment (bscheltinga) [reply scheduler/state_machine/state_machine_utils.py:788]: 🙏

**Pull request merged**